autogpt/autogpt/agents/agent.py

```diff
@@ -116,27 +120,33 @@ def __init__(
         self.commands: list[Command] = []
 
         # Components
-        self.system = SystemComponent(legacy_config, settings.ai_profile)
+        self.system = SystemComponent(settings.ai_profile)
         self.history = ActionHistoryComponent(
             settings.history,
-            self.send_token_limit,
             lambda x: self.llm_provider.count_tokens(x, self.llm.name),
-            legacy_config,
+            legacy_config.fast_llm,
```

Comment:
Functionally this means the LLM used by `ActionHistoryComponent` can no longer change at runtime. Not that we're doing that right now, I just wanted to note this explicitly because it could become relevant in the near future.

Reply:
Moved `model_name` inside config for `ActionHistoryComponent` and `WebSeleniumComponent`.